sqflite/pubspec.yaml

```diff
@@ -24,4 +24,4 @@ dev_dependencies:
   mockito: ^5.1.0
   build_runner: ^2.4.2
   yaml: ^3.1.0 # needed for version match (code and pubspec)
-  sqflite_common_ffi: ^2.0.0
+  sqflite_common_ffi: ^2.3.0
```

Comment:
This is a breaking change, if intentional, ideally another PR?

Reply:
Will also revert, it's a bit tricky as `sqflite` introduced the new method `readTransaction` in their [most recent update](https://pub.dev/packages/sqflite_common/changelog#2502), which we then need to override, but compiler of course complains for versions between 2.0.0 and the most recent one when using override.

So as a solution I just tried this approach:
```dart
// ignore: annotate_overrides
Future<T> readTransaction<T>(Future<T> Function(Transaction txn) action) {
  // Use fallback that is available in al 2.X versions of sqflite.
  return _database.transaction(action);
}
```

It's an experimental method and we can use 'transaction' instead. That way we don't have to update to `2.5.0+2` as the minimum version & break support.

Reply:
You can keep compatibility with both versions by just overriding the new method and adding `// ignore: override_on_non_overriding_member`
we did that plenty of times, so its ignored when using < supported version and used when its >= supported version, so it compiles against 2.0.0 as well

Reply:
I'd do that in a separate PR though, merges first, and this one won't fail anymore

Reply:
Ahh, sounds good. I'll create a sep. PR and use `// ignore: override_on_non_overriding_member` thanks! 🙇